backtracking/n_queens.cpp

```diff
@@ -1,63 +1,129 @@
+/**
+ * @file n_queens.cpp
```

Comment:
```suggestion
 * @file
```
Better to remove the filename

Reply:
Done, removed.